src/Database/Table/ReferenceCache.php

```diff
@@ -0,0 +1,65 @@
+<?php
+
+/**
+ * This file is part of the Nette Framework (https://nette.org)
+ * Copyright (c) 2004 David Grudl (https://davidgrudl.com)
+ */
+
+declare(strict_types=1);
+
+namespace Nette\Database\Table;
+
+
+class ReferenceCache
+{
+	/** @var array */
+	protected $referencingPrototype = [];
+
+	/** @var array */
+	protected $referencing = [];
+
+	/** @var array */
+	protected $referenced = [];
+
+
+	public function &getReferencingPrototype(string $specificCacheKey, string $table, string $column)
+	{
+		return $this->referencingPrototype[$specificCacheKey]["$table.$column"];
+	}
+
+
+	public function clearReferencingPrototype(): void
+	{
+		$this->referencingPrototype = [];
+	}
+
+
+	/**
+	 * @return array|null
+	 */
+	public function &getReferencing(string $generalCacheKey)
```

Comment:
Sry, now I realized that it can be written native as `&getReferencing(string $generalCacheKey): ?array`

Reply:
Yes, I should realize that too. Fixed